web/utils_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build unit
+
+package web
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/control-center/serviced/zzk/registry"
+	"github.com/control-center/serviced/zzk/service"
+	. "gopkg.in/check.v1"
+)
+
+var (
+	privateIp string = "private-ip"
+	privatePort uint16 = 12345
+	hostIp  string = "host-ip"
+	hostPort uint16 = 22250
+	
+	serviceAddress string = fmt.Sprintf("%s:%d", privateIp, privatePort)
+	muxAddress string = fmt.Sprintf("%s:%d", hostIp, hostPort)	
+)
+
+type dialedServiceAddress struct{}
+func (e *dialedServiceAddress) Error() string {
+	return serviceAddress
+}
+
+type dialedMuxAddress struct{}
+func (e *dialedMuxAddress) Error() string {
+	return muxAddress
+}
+
+// Mock dialers that just return an error on dial based on the address given.
+type mockDialer struct{}
```

Comment:
This is all a little brute force. Here's how to replace this impl with the GO mocking lib we use:
```
import (
	"net"
	"github.com/stretchr/testify/mock"
)

type mockDialer struct {
	mock.Mock
}

func (_m *mockDialer) Dial(network, address string) (net.Conn, error)  {
	ret := _m.Called(d)

	var r0 net.Conn
	if rf, ok := ret.Get(0).(func(string, string) net.Conn); ok {
		r0 = rf(network, address)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(net.Conn)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(string, string) error); ok {
		r1 = rf(network, address)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}
```